File: JavaProject/Hotel.java
import java.util.Date;
public class Hotel {
	
	private Employees employees;
	private Clients clients;
	private Documentation documentation;
	private Reservations reservations;
	private Rooms rooms;
	private Client fitting_client;
	private Room[] fitting_rooms;
	public void Make_reservation(String client_id, Date start_date, Date end_date, int room_number){
		
	}
	public void Take_payment(String client_id, double value, boolean card){
		
	}
	public void Register_client(String last_name, String first_name, String client_id, String email){
		
	}
	public Room[] Find_free_room(Date start_date, Date end_date, int capacity){
		
		return fitting_rooms;
	}
	public void Check_in_client(String reservation_id){
		
	}
	public void Cancel_reservation(String reservation_id){
		
	}
	public void Check_out_client(String reservation_id){
		
	}
	public Client Find_client(String client_id){

		return fitting_client;
	}
	public void Call_menager(String place){
		
	}
	public void report_need_for_cleaning(int room_number){
		
	}
	public void report_need_for_repair(int room_number){
		
	}
	public void report_need_for_security_intervention(int room_number){
		
	}
	public void Report_execution_of_cleaning(int room_number){
		
	}
	public void Report_execution_of_repair(int room_number){
		
	}
	public void View_reports(){
		
	}
	public void Print_and_archive_report(Date date){
		
	}
	public void Add_daily_report(){
		
	}
	public void PayOut_salary(String employee_ID){
		
	}
	public void Give_bonus(String employee_id, double bonus){
		
	}
	public void Calculate_salary(String employee_id){
		
	}
	public void Fire_employee(String employee_id){
		
	}
	public void Check_hotel_statistics(){
		
	}
	public void Call_employee(String employee_id){
		
	}
	public void Warn_employee(String employee_id){
		
	}
	public Employees getEmployees() {
		return employees;
	}
	
	public void setEmployees(Employees employees) {
		this.employees = employees;
	}
	
	public Clients getClients() {
		return clients;
	}
	
	public void setClients(Clients clients) {
		this.clients = clients;
	}
	
	public Documentation getDocumentation() {
		return documentation;
	}
	
	public void setDocumentation(Documentation documentation) {
		this.documentation = documentation;
	}
	
	public Reservations getReservations() {
		return reservations;
	}
	
	public void setReservations(Reservations reservations) {
		this.reservations = reservations;
	}
	
	public Rooms getRooms() {
		return rooms;
	}
	
	public void setRooms(Rooms rooms) {
		this.rooms = rooms;
	}


}

File: CSharpProject/Hotel/Model/Hotel.cs
using Hotel.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HotelProject.Model
{
    public class Hotel
    {
        Employees employees;
        Clients clients;
        Documentation documentation;
        Reservations reservations;
        Rooms rooms;

        public Hotel()
        {
            employees = new Employees();
            clients = new Clients();
            documentation = new Documentation();
            reservations = new Reservations();
            rooms = new Rooms();
        }

        public void MakeReservation(String clientID, DateTime startDate, DateTime endDate, int roomNumber)
        {
            throw new NotImplementedException();
        }

        public void RegisterClient(String firstName, String lastName, String clientID, String email = "")
        {
            throw new NotImplementedException();
        }

        public int[] FindFreeRoom(DateTime startDate, DateTime endDate, int capacity)
        {
            return reservations.FindFreeRoom(startDate, endDate, capacity, rooms);
        }

        public void CheckInClient(String reservationID, IRooms roomsCollection)
        {
            throw new NotImplementedException();
        }

        public void CancelReservation(String clientID)
        {
            throw new NotImplementedException();
        }

        public void CheckOutClient(int roomNumber)
        {
            throw new NotImplementedException();
        }

        public void FindClient(String clientID)
        {
            throw new NotImplementedException();
        }

        public void CallManager(String place)
        {
            throw new NotImplementedException();
        }

        public void ReportNeedForCleaning(int roomNumber)
        {
            rooms.ReportNeedForCleaning(roomNumber);
        }

        public void ReportNeedForRepair(int roomNumber)
        {
            rooms.ReportNeedForRepair(roomNumber);
        }

        public void ReportNeedForSecurityIntervention(int roomNumber)
        {
            throw new NotImplementedException();
        }

        public void TakePayment(String clientID, double paymentValue, bool card)
        {
            throw new NotImplementedException();
        }

        public void ReportExecutionOfRepair(int roomNumber, bool occupied)
        {
            rooms.ReportExecutionOfRepair(roomNumber, occupied);
        }

        public void ReportExecutionOfCleaning(int roomNumber, bool occupied)
        {
            rooms.ReportExecutionOfCleaning(roomNumber, occupied);
        }

        public void ViewReports()
        {
            throw new NotImplementedException();
        }

        public void PrintAndArchive(DateTime date)
        {
            throw new NotImplementedException();
        }

        public void AddDailyReport()
        {
            throw new NotImplementedException();
        }

        public void PayOutSalary(String employeeID)
        {
            throw new NotImplementedException();
        }

        public void GiveBonus(String employeeID)
        {
            throw new NotImplementedException();
        }

        public void CalculateSalary(String employeeID)
        {
            throw new NotImplementedException();
        }

        public void FireEmployee(String employeeID)
        {
            throw new NotImplementedException();
        }

        public void CallEmployee(String employeeID)
        {
            throw new NotImplementedException();
        }

        public void WarnEmployee(String employeeID)
        {
            throw new NotImplementedException();
        }
    }
}
